pkg/gateway/model_build_access_log_subscription.go

```diff
@@ -72,9 +72,31 @@ func (t *accessLogSubscriptionModelBuildTask) run(ctx context.Context) error {
 		return fmt.Errorf("access log policy's destinationArn cannot be nil")
 	}
 
-	isDeleted := t.accessLogPolicy.DeletionTimestamp != nil
+	var eventType = core.CreateEvent
+	if t.accessLogPolicy.DeletionTimestamp != nil {
+		eventType = core.DeleteEvent
+	} else if _, ok := t.accessLogPolicy.Annotations[anv1alpha1.AccessLogSubscriptionAnnotationKey]; ok {
+		eventType = core.UpdateEvent
+	}
+
+	var status *model.AccessLogSubscriptionStatus
+	if eventType != core.CreateEvent {
+		value, exists := t.accessLogPolicy.Annotations[anv1alpha1.AccessLogSubscriptionAnnotationKey]
+		if exists {
+			status = &model.AccessLogSubscriptionStatus{
+				Arn: value,
+			}
+		}
```

Comment:
if it's not CreateEvent the value should always exists, right? should we throw error here or log?

Reply:
Will throw here, good point

Reply:
Correction, this is allowed to not exist because Creation might fail, resulting in annotation not being added, and user might try to delete the ALP. I'll add a debug log, rather than throw.